packages/run-protocol/src/vaultFactory/vaultDirector.js

```diff
@@ -74,260 +81,296 @@ const initState = (zcf, electorateParamManager, debtMint) => {
-          // stolen here because the staging for them was already cleared.
-          debtMint.burnLosses(harden({ RUN: toMint }), mintSeat);
-          throw e;
-        } finally {
-          assert(
-            Object.values(mintSeat.getCurrentAllocation()).every(a =>
-              AmountMath.isEmpty(a),
-            ),
-            X`Stage should be empty of RUN`,
-          );
-        }
-        // TODO add aggregate debt tracking at the vaultFactory level #4482
-        // totalDebt = AmountMath.add(totalDebt, toMint);
-      };
+    /**
+     * We provide an easy way for the vaultManager to add rewards to
+     * the rewardPoolSeat, without directly exposing the rewardPoolSeat to them.
+     *
+     * @type {MintAndReallocate}
+     */
+    const mintAndReallocate = (toMint, fee, seat, ...otherSeats) => {
+      const kept = AmountMath.subtract(toMint, fee);
+      debtMint.mintGains(harden({ RUN: toMint }), mintSeat);
+      try {
+        rewardPoolSeat.incrementBy(mintSeat.decrementBy(harden({ RUN: fee })));
+        seat.incrementBy(mintSeat.decrementBy(harden({ RUN: kept })));
+        zcf.reallocate(rewardPoolSeat, mintSeat, seat, ...otherSeats);
+      } catch (e) {
+        mintSeat.clear();
+        rewardPoolSeat.clear();
+        // Make best efforts to burn the newly minted tokens, for hygiene.
+        // That only relies on the internal mint, so it cannot fail without
+        // there being much larger problems. There's no risk of tokens being
+        // stolen here because the staging for them was already cleared.
+        debtMint.burnLosses(harden({ RUN: toMint }), mintSeat);
+        throw e;
+      } finally {
+        assert(
+          Object.values(mintSeat.getCurrentAllocation()).every(a =>
+            AmountMath.isEmpty(a),
+          ),
+          X`Stage should be empty of RUN`,
+        );
+      }
+      // TODO add aggregate debt tracking at the vaultFactory level #4482
+      // totalDebt = AmountMath.add(totalDebt, toMint);
+    };
 
-      /**
-       * @param {Amount<'nat'>} toBurn
-       * @param {ZCFSeat} seat
-       */
-      const burnDebt = (toBurn, seat) => {
-        debtMint.burnLosses(harden({ RUN: toBurn }), seat);
-      };
+    /**
+     * @param {Amount<'nat'>} toBurn
+     * @param {ZCFSeat} seat
+     */
+    const burnDebt = (toBurn, seat) => {
+      debtMint.burnLosses(harden({ RUN: toBurn }), seat);
+    };
 
-      const { loanTimingParams } = zcf.getTerms();
+    const { loanTimingParams } = zcf.getTerms();
 
-      const factoryPowers = Far('vault factory powers', {
-        getGovernedParams: () => ({
-          ...vaultParamManager.readonly(),
-          getChargingPeriod: () => loanTimingParams[CHARGING_PERIOD_KEY].value,
-          getRecordingPeriod: () =>
-            loanTimingParams[RECORDING_PERIOD_KEY].value,
-        }),
-        mintAndReallocate,
-        burnDebt,
-      });
+    const factoryPowers = Far('vault factory powers', {
```

Comment:
will defining a far per vault manager pose any problems for durability @FUDCo ?

Reply:
On general principles it should not be a problem, but the key question is: will you be able to reconstruct the Far from the durable state after resurrection?  (I can't discern the answer to that myself that without digging into this code considerably beyond my feeble understanding of what it's doing.)  If you can, then it should be no problem; you just need to make sure your startup flow actually does it.  Otherwise you'll either need to (a) capture more info in the durable part so that you _can_ reconstruct it or (b) make it durable itself.  I don't have an opinion which is more right, as it would depend on the particulars, other than the general advice to take whichever course causes you the least pain.